### DISCUSSION TITLE
How to obtain preview gif from review event

### DISCUSSION THREAD

**Discussion by leonbrag** [blakeblackshear/frigate#13590] [General]
**Objective:** 
- obtain short looping video from reviews event
- exactly what is returned by GET /api/events/id/preview.gif

**Problem:**
Because I subscribe to reviews MQTT topic and not events, I don't have event id to use in the API above.
Hence I trying to use GET /<camera>/start/start-timestamp/end/end-timestamp/preview?format=gif without success.
I will then attach to email.
**Details**

This is a valid review event:

http://hpdebian:5000/api/review/1725639992.022836-blrrq9
```
{
    "camera": "Back_Yard",
    "data": {
        "audio": [],
        "detections": [
            "1725639991.424938-uvh2o8"
        ],
        "objects": [
            "person"
        ],
        "sub_labels": [],
        "zones": []
    },
    "end_time": 1725639998.825952,
    "has_been_reviewed": false,
    "id": "1725639992.022836-blrrq9",
    "severity": "alert",
    "start_time": 1725639992.022836,
    "thumb_path": "/media/frigate/clips/review/thumb-Back_Yard-1725639992.022836-blrrq9.webp"
}
```

However when I form URL using timestamps from event above: 

http://hpdebian:5000/api/Back_Yard/start/1725639992.022836/end/1725639998.825952/preview?format=gif

I get "Not Found error". Note that URL http://hpdebian:5000/api/Back_Yard is valid.

Am I doing something obviously wrong? 
Is there easier way to get looping video from review ?

Thanks

**Comment by NickM-27** [comment]
1. You do have effect ids. `detections`is just a list of event ids
2. There is also a `/api/review/id/preview.gif` api

**Reply by leonbrag** [reply]
Thanks!! 
#1 worked perfectly,  I did not realize that detections array are just events. 
#2 did not work, but #1 is all I need.
```
http://hpdebian:5000/api/review/1725723703.74136-g7uowj
{
    "camera": "Front_Door",
    "data": {
        "audio": [],
        "detections": [
            "1725723983.838824-2rab77",
            "1725723731.726872-umeds8",
            "1725723876.743665-gtiyxv",
            "1725723727.238024-clkrhb",
            "1725723592.69644-47ela2",
            "1725723906.04765-h4atjk",
            "1725723757.431861-vt87jj"
        ],
        "objects": [
            "person"
        ],
        "sub_labels": [],
        "zones": [
            "frontYard"
        ]
    },
    "end_time": 1725724003.711225,
    "has_been_reviewed": false,
    "id": "1725723703.74136-g7uowj",
    "severity": "alert",
    "start_time": 1725723703.74136,
    "thumb_path": "/media/frigate/clips/review/thumb-Front_Door-1725723703.74136-g7uowj.webp"
}
     
```   
        Using ID of the Review event from above:
        
        GET http://hpdebian:5000/api/review/1725723703.74136-g7uowj/preview.gif 404 (NOT FOUND)

**Reply by NickM-27** [reply]
I did not realize that you were connecting directly, thought it was through the integration. It is `/api/review/<id>/preview` (no .gif because you can request different types)

**Reply by leonbrag** [reply]
yep. It sure worked, thanks!
Is it not documented in preview section by design? Will API stay there in the future ?

![image](https://github.com/user-attachments/assets/ec9fa343-d9b7-431d-903b-8f1f080c9491)

**Reply by NickM-27** [reply]
oh, it was just added late and I guess we forgot to add it

**Reply by leonbrag** [reply]
Thanks. I took liberty to propose doc changes & created PR: https://github.com/blakeblackshear/frigate/pull/13614

**Comment by leonbrag** [comment]
solved
